### PR TITLE
Move library to TypeScript

### PR DESCRIPTION
**Pull request by mrruby**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,22 +1,26 @@
-name: hcSeedBundle Test
+name: @holochain/hc-seed-bundle Test and Release
 
 on:
   push:
     branches: [ main ]
-  pull_request:
-    branches: [ main ]
```
Comment (neonphog) [inline .github/workflows/ci.yaml:7]: This appears to no longer be running unit tests on pull requests. In general we like to see passing unit tests before merging.

File: bun.lockb
Comment (neonphog) [inline bun.lockb:1]: Is it correct that this `bun.lockb` file is checked in?

**Review by neonphog — CHANGES_REQUESTED** [review]
@mrruby 

The translation looks good, and I'm glad it looks like the tests translated fairly directly. I've never used bun before, but if that works with your workflow, I'm happy for it to be used in this repo.

Some things I think we need to change before merging:
- I don't know what a `bun.lockb` is... just double check it's supposed to be in the repo
- get unit tests running before PR merge
- add back in doc comments throughout
- decide what to do with documentation generation:
  - either decide typescript doesn't really need html form documentation, in which case we need to remove all the references to it including in the README.md and on the side bar in github, or
  - add in typedoc docs so they can take the place of the jsdocs

File: package.json [hunk 2]
```diff
@@ -13,33 +13,36 @@
     "type": "git",
     "url": "github:holochain/hcSeedBundle.git"
   },
-  "main": "index.js",
-  "type": "module",
   "license": "Apache-2.0",
-  "scripts": {
-    "lint": "standard",
-    "mocha": "mocha",
-    "doc": "jsdoc -d docs -R README.md index.js",
```
[the remark this answers is not in the document]
Comment (mrruby) [reply package.json:22]: added

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (mrruby) [reply .github/workflows/ci.yaml:7]: fixed

File: index.ts [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+    }
+    this.passphrase = passphrase;
+    this.limitName = limitName;
+  }
+
+  zero(): void {
+    this.passphrase.zero();
+  }
+
+  encryptSeed(secretSeed: PrivSecretBuf): object {
```
[the remark this answers is not in the document]
Comment (mrruby) [reply index.ts:273]: fixed

**Review by neonphog — APPROVED** [review]
Nice, lgtm!

**Pull request merged**